File: createams-back/src/main/java/com/myproject/DTO/UserDTO.java
package com.myproject.DTO;

import lombok.Data;

import java.util.List;

@Data
public class UserDTO {
    private Long id;
    private String pseudo;
    private String mail;
    private String surname;
    private String name;
    private Boolean premium;
    private List<ProjectDTO> projects;
    private String job;
    private String image;
    private String mentor;
}

File: createams-back-node-methode/src/user.dto.ts
export interface ProjectDTO {
    pseudo?: string;
    project_name?: string;
    users_id?: string;
    description?: string;
}

export interface UserDTO {
    pseudo?: string;
    mail?: string;
    surname?: string;
    name?: string;
    premium?: boolean;
    project?: ProjectDTO;
}